vscode-trace-extension/src/utils/trace-extension-event-manager.ts

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************************
+ * Copyright (c) 2023 BlackBerry Limited and others.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+import * as vscode from 'vscode';
+
+export interface SignalCallbackInterface {
+    signal: string,
+    callback: (data: unknown) => void
+}
+
+/**
+ *  Manages the events for the trace extension
+ */
+export class TraceExtensionEventManager {
+    private eventEmitterMap: Map<string, vscode.EventEmitter<unknown>>;
+    constructor() {
+        this.eventEmitterMap = new Map<string, vscode.EventEmitter<unknown>>();
+    }
+
+    /**
+     * Get the event emitter registered based on Id
+     *
+     * @param id event emitter Id
+     * @returns event emitter for the given id if exists, otherwise undefined
+     */
+    getEventEmitter(id: string): vscode.EventEmitter<unknown> | undefined {
+        return this.eventEmitterMap.get(id);
+    }
+
+    /**
+     * Register an event emitter with the provided Id
+     *
+     * @param id event emitter Id
+     */
+    registerEventEmitter(id: string): void {
+        this.eventEmitterMap.set(id, new vscode.EventEmitter<unknown>());
+    }
+
+    /**
+     * Fire the event emitter with the provided Id
+     *
+     * @param id event emitter Id
+     * @param data event object
+     */
+    fireEventEmitter(id: string, data: unknown): void {
+        this.getEventEmitter(id)?.fire(data);
+    }
+
+    /**
+     * Retrieves an iterable event emitter Ids
+     *
+     * @returns Iterable Event Ids
+     */
+    getKeys(): Iterable<string> {
+        return this.eventEmitterMap.keys();
+    }
+
+    /**
+     * Attaches a listener for an event based on the event emitter id
+     *
+     * @param id event emitter Id
+     * @param callback function to call when the event is triggered
+     */
+    onEvent(id: string, callback: (data: unknown) => void): void {
+        this.getEventEmitter(id)?.event(callback);
+    }
```

Comment:
Is there a reason not to return the `Disposable` that is returned by the underlying `event` call?

And if the emitter doesn't currently exist, should it be created? Or should the caller at least be informed in some way that their subscription failed?